src/subsys/ngx_subsys_lua_util.c.tt2

```diff
@@ -828,9 +828,11 @@ ngx_[% subsys %]_lua_inject_ngx_api(lua_State *L, ngx_[% subsys %]_lua_main_conf
                 "if phase == 'init_worker' or phase == 'init' then\n"
                     "return\n"
                 "end\n"
-                "ngx_log(ngx_WARN, '[% subsys %] lua attempting to write to the "
-                         "lua global variable \\'', tostring(key), "
-                         "'\\'\\n', traceback())\n"
+                "ngx_log(ngx_WARN, 'writing a global lua variable "
+                         "(\\'', tostring(key), '\\') which may lead to "
+                         "race conditions between concurrent requests, so "
+                         "prefer the use of \\'local\\' variables', "
+                         "traceback('', 2))\n"
```

Comment:
The empty string (`''`) given to `debug.traceback` is required when giving the function a second argument (the stack level).